app/services/cloud_front_request.rb

```diff
@@ -0,0 +1,61 @@
+class CloudFrontRequest
```

Comment:
Did I go too far off the beaten path here?

I didn't want to try to parse port number out of the header in tracker.rb, so I figured I'd implement a little class for it, and then I figured it might as well cover all the new CloudFront headers.

Reply:
This makes sense! I'd consider calling it `CloudFrontHeadersParser` or something like that but otherwise same same